File: core/src/com/qtech/bubbleblaster/event/registry/RegistryEvent.java
package com.qtech.bubbleblaster.event.registry;

import com.qtech.bubbleblaster.common.IRegistryEntry;
import com.qtech.bubbleblaster.event.Event;
import com.qtech.bubbleblaster.registry.Registry;

import java.util.Objects;

public class RegistryEvent {
    public static class Register<T extends IRegistryEntry> extends Event {
        private final Registry<T> registry;

        public Register(Registry<T> registry) {
            super();
            this.registry = registry;
        }

        public Registry<T> getRegistry() {
            return registry;
        }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (o == null || getClass() != o.getClass()) return false;
            Register<?> register = (Register<?>) o;
            return Objects.equals(getRegistry().getType(), register.getRegistry().getType());
        }

        @Override
        public int hashCode() {
            return Objects.hash(getRegistry().getType());
        }
    }

    public static class Dump extends Event {
        public Dump() {
            super();
        }
    }
}

File: src/main/kotlin/qtech/bubbles/event/registry/RegistryEvent.kt
package qtech.bubbles.event.registry

import qtech.bubbles.common.IRegistryEntry
import qtech.bubbles.event.Event
import qtech.bubbles.registry.Registry
import java.util.*

class RegistryEvent {
    class Register<T : IRegistryEntry>(val registry: Registry<T>) : Event() {
        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (other == null || javaClass != other.javaClass) return false
            val register = other as Register<*>
            return registry.type == register.registry.type
        }

        override fun hashCode(): Int {
            return Objects.hash(registry.type)
        }
    }

    class Dump : Event()
}